hs_modflow_modelinstance/receivers.py

```diff
@@ -89,3 +92,232 @@ def _process_metadata_update_create(update_or_create, **kwargs):
+                    continue
+                if p in modflow_models.uncouple(
+                        modflow_models.BoundaryCondition.specifiedFluxBoundaryPackageChoices):
+                    # create if does not exist, update if it does exist
+                    specified_flux_boundary_packages_list.append(p)
+                    create_or_update_from_package(
+                        resource, modflow_models.BoundaryCondition,
+                        specified_flux_boundary_packages=specified_flux_boundary_packages_list)
+                    continue
+                if p in modflow_models.uncouple(
+                        modflow_models.BoundaryCondition.headDependentFluxBoundaryPackageChoices):
+                    # create if does not exist, update if it does exist
+                    head_dependent_flux_boundary_packages.append(p)
+                    create_or_update_from_package(
+                        resource, modflow_models.BoundaryCondition,
+                        head_dependent_flux_boundary_packages=head_dependent_flux_boundary_packages)
+                    continue
+                # ModelCalibration
+                if p in modflow_models.uncouple(
+                        modflow_models.ModelCalibration.observationProcessPackageChoices):
+                    # create if does not exist, update if it does exist
+                    create_or_update_from_package(resource, modflow_models.ModelCalibration,
+                                                  observationProcessPackage=p)
+                    continue
+                # GeneralElements
+                if p in modflow_models.uncouple(
+                        modflow_models.GeneralElements.modelSolverChoices):
+                    # create if does not exist, update if it does exist
+                    create_or_update_from_package(resource, modflow_models.GeneralElements,
+                                                  modelSolver=p)
+                    continue
+                if p in modflow_models.uncouple(
+                        modflow_models.GeneralElements.outputControlPackageChoices):
+                    # create if does not exist, update if it does exist
+                    output_control_package_list.append(p)
+                    create_or_update_from_package(
+                        resource, modflow_models.GeneralElements,
+                        output_control_package=output_control_package_list)
+                    continue
+                if p in modflow_models.uncouple(
+                        modflow_models.GeneralElements.subsidencePackageChoices):
+                    # create if does not exist, update if it does exist
+                    create_or_update_from_package(resource, modflow_models.GeneralElements,
+                                                  subsidencePackage=p)
+                    continue
+
+
+def create_or_update_from_package(resource, term, **kwargs):
+    if term.term == 'StressPeriod':

```

Comment:
In place all these ```if``` statements, you can create a dictionary (say terms_dict) with key as term.term and corresponding value for t. Then you can do:
```t = terms_dict(term.term)```